### PR TITLE
Create Examples of functionnal tests

### PR DESCRIPTION
**Pull request by Lee31416**
GitHub Issue: #

## Proposed Changes

 - [ ] Bug fix
 - [x] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

Adding examples of functionnal tests that run on the pipeline. 

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [x] **Minor**
  - New functionalities were added.
- [ ] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [x] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [x] Automated tests for the changes have been added/updated.
- [x] Tested on all relevant platforms

### CODE REVIEW

File: src/app/lib/business/dad_jokes/dad_jokes_service.dart [hunk 1]
```diff
@@ -11,8 +11,8 @@ import 'package:rxdart/rxdart.dart';
 /// Interface for the dad jokes service.
 abstract interface class DadJokesService implements Disposable {
   factory DadJokesService(
-    DadJokesRepository dadJokesRepository,
-    FavoriteDadJokesRepository favoriteDadJokesRepository,
+    DadJokesRepositoryDecorator dadJokesRepository,
```
Comment (Soap-141) [inline src/app/lib/business/dad_jokes/dad_jokes_service.dart:14]: This should probably stay like it was, but the decorator should implement the interface and have the inner service. Like on one of our internal projects right?

File: src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart [hunk 2]
```diff
@@ -21,13 +22,14 @@ class _ExpandedDiagnosticPageState extends State<ExpandedDiagnosticPage>
...
     LoggerDiagnosticWidget(),
+    const MockingDiagnosticWidget(),
   ];
 
   int _selectedIndex = 0;
 
   @override
   void initState() {
-    _tabController = TabController(length: 4, vsync: this);
+    _tabController = TabController(length: 5, vsync: this);
```
Comment (Soap-141) [inline src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart:32]: Can't we use `expandedDiagnosticWidgets.length` here? Or something similar?

File: src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import 'package:app/presentation/diagnostic/device_info_widget.dart';
 import 'package:app/presentation/diagnostic/environment_diagnostic_widget.dart';
 import 'package:app/presentation/diagnostic/logger_diagnostic_widget.dart';
+import 'package:app/presentation/diagnostic/mocking_diagnostic_widget.dart';
 import 'package:app/presentation/diagnostic/navigation_diagnostic_widget.dart';
 import 'package:flutter/material.dart';
```
Comment (Soap-141) [inline src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart:7]: The state can probably be final too.

File: CHANGELOG.md [hunk 4]
```diff
@@ -33,3 +33,5 @@ Prefix your items with `(Template)` if the change is about the template and not
 - Added Alice as a tool to see logs in app.
 - Fix diagnostic overlay dismiss behavior so it's not permanent.
 - Added test for the UpdateRequiredService and fixed bug where remote config was not updating correctly.
+
```
[the remark this answers is not in the document]
Comment (Lee31416) [reply CHANGELOG.md:36]: Those damned merge sometimes they merge with the whitespace of the incoming/current and I miss it

File: src/app/lib/business/dad_jokes/dad_jokes_service.dart (hunk 1 above)
Comment (Lee31416) [reply src/app/lib/business/dad_jokes/dad_jokes_service.dart:14]: I was under the impression that flutter wouldn't recognize the decorator as the repository because they don't extend they implement the interface but it seems the service understands that they have the same parent so you was right and I learned something lol

File: doc/Testing.md [hunk 5]
```diff
@@ -67,6 +67,34 @@ For more documentation on testing, read the references listed at the bottom.
...
+		final MockingRepository _mockingRepository;
+
+		DadJokesRepositoryDecorator(this._innerRepository, this._mockingRepository);
+
+		@override
+		Future<DadJokeResponseData> getDadJokes() async {
+			var isMockingEnabled = await _mockingRepository.checkMockingEnabled();
+
+			if (isMockingEnabled) {
+				return mockedDadJokeResponse;
```
Comment (jeanplevesque) [inline doc/Testing.md:84]: I don't know if it's important, but the variable `mockedDadJokeResponse` doesn't seem to exist.

File: doc/Testing.md [hunk 6]
```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
     }
     ```
 
+## Functionnal/Integration Testing
+
+- We use [integration_test](https://docs.flutter.dev/testing/integration-tests) to create functionnal/integration tests. You can create a test like this :
```
Comment (jeanplevesque) [inline doc/Testing.md:131]: Write integration_test**s** if that's the name of the package

File: doc/Testing.md [hunk 7]
```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
...
+	testWidgets('Get Dad Jokes', (WidgetTester tester) async {
+		// Arrange
+		await tester.pumpWidget(const App());
+
+		// Without this the dadjokesContainer isn't there yet.
+		await tester.pumpAndSettle();
+
+		var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+		// Act
```
Comment (jeanplevesque) [inline doc/Testing.md:148]: I'm not sure this is actually an **Act** because you're just fetching some widgets. It feels more like part of the **Assert**.
To me the Act of this test is simply the creation of the App.

File: doc/Testing.md [hunk 8]
```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
...
+  await tester.tap(dadJokes.first);
+  ```
+
+- To assert the result of a test, use `expect()` exactly like with unit tests.
+
+- tester.pumpAndSettle() is used both to trigger a frame change and to wait for the last pump to have settled before moving on. For example, we use it after pumping the app widget and we also use it when we naviguated and we want to update the UI.
+
+### Mocking
+
+For functionnal testing we use the decorator pattern because we are testing the actual behavior of the app so we don't wanna use mockito to mock the logic of the classes of the app. To use the decorator pattern you simply call the mocking repository and the method called setMocking(true). It's important to set the mocking in the main function before the tests because otherwise the mocking doesn't take effect. In most cases, except for api tests, the data sources(repositories) should be mocked for integration testing.
```
Comment (jeanplevesque) [inline doc/Testing.md:171]: Add a space after data sources before the parenthesis.

File: src/app/lib/access/mocking/mocking_repository.dart [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+abstract interface class MockingRepository {
+  factory MockingRepository() = _MockingRepository;
+
+  /// Gets whether the mocking is enabled.
+  Future<bool> checkMockingEnabled();
```
Comment (jeanplevesque) [inline src/app/lib/access/mocking/mocking_repository.dart:7]: `checkMockingEnabled` doesn't seem like a conventional name for a method that gets a value.

File: src/app/lib/business/dad_jokes/dad_jokes_service.dart (hunk 1 above)
Comment (jeanplevesque) [reply src/app/lib/business/dad_jokes/dad_jokes_service.dart:14]: Yeah the business service must not know about the specific implementations. It needs to use only the abstractions. Otherwise it gets tightly coupled to a specific implementation and breaks the possibility of substitution.

File: src/app/lib/main.dart [hunk 10]
```diff
@@ -29,6 +33,12 @@ import 'package:logger/logger.dart';
 late Logger _logger;
 
 Future<void> main() async {
+  await initializeComponents();
+
+  runApp(const App());
+}
+
+Future initializeComponents({bool? overrideIsMocked}) async {
```
Comment (jeanplevesque) [inline src/app/lib/main.dart:41]: I think this should be renamed to `isMocked` to make it clearer.
- null keeps the default behavior
- true forces mock data
- false forces real data

Having the parameter named overrideXXX sounds a bit more confusing around what is the boolean about.

File: src/app/test/business/dad_jokes_service_test.dart [hunk 11]
```diff
@@ -1,49 +1,52 @@
...
-  ],
-)
+
 void main() {
-  late DadJokesRepository dadJokesRepository;
-  late FavoriteDadJokesRepository favoriteDadJokesRepository;
+  late DadJokesRepositoryDecorator dadJokesRepository;
+  late FavoriteDadJokesRepositoryDecorator favoriteDadJokesRepository;
+  late MockingRepository mockingRepository;
   late DadJokesService SUT;
```
[the remark this answers is not in the document]
Comment (jeanplevesque) [reply src/app/test/business/dad_jokes_service_test.dart:23]: subject under test

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: doc/Testing.md [hunk 12]
```diff
@@ -67,6 +67,34 @@ For more documentation on testing, read the references listed at the bottom.
...
+			if (isMockingEnabled) {
+				return mockedDadJokeResponse;
+			} else {
+				return _innerRepository.getDadJokes();
+			}
+		}
+	}
+	``` 
+
+	Then we simply call :
```
Comment (takla21) [inline doc/Testing.md:92]: I would probably reword it something like, `Then, enable mock by calling this method`

File: doc/Testing.md [hunk 13]
```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
...
+		await tester.pumpWidget(const App());
+
+		// Without this the dadjokesContainer isn't there yet.
+		await tester.pumpAndSettle();
+
+		var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+		// Act
+		var dadJokes = find.descendant(
+		of: dadjokesContainer,
```
Comment (takla21) [inline doc/Testing.md:150]: nitpick: indent parameters like this;

```cs
var dadJokes = find.descendant(
	of: dadjokesContainer,
	matching: find.byType(DadJokeListItem),
);
```

File: doc/Testing.md (hunk 8 above)
Comment (takla21) [inline doc/Testing.md:171]: I would reword your third sentence here like this

It's important to set the mocking in the main function before the tests otherwise the mocking doesn't take effect.

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Marc-Antoine-Soucy) [reply CHANGELOG.md:36]: actually, it's better to leave the blank line

[the remark this answers is not in the document]
Comment (Marc-Antoine-Soucy) [reply CHANGELOG.md:36]: You should still leave a blank line at the end of the .md file thought.

File: doc/Testing.md [hunk 14]
```diff
@@ -98,6 +126,60 @@ For more documentation on testing, read the references listed at the bottom.
...
+
+### Running the tests
+
+- You can run the integration test with this command :
+```bash
+flutter test integration_test --dart-define ENV=Development
+```
+  We need to set the env because the environment manager doesn't set one by default when we load the app it is set for each build accordingly so we need to manually set it. 
+
+- When testing locally it is important that the device on which you are testing is opened and running because the tests will interact with the app in real time. Otherwise it will throw unclear exceptions that don't point you at this issue.
```
Comment (takla21) [inline doc/Testing.md:181]: I personally didn't like how the first sentence was worded. Therefore, I've regenerated that first sentence with our friends ChatGPT & Grammarly `When testing locally, the device on which you are testing must be open and running, as the tests will interact with the app in real-time.`

File: src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart (hunk 2 above)
Comment (takla21) [reply src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart:32]: that's a good idea

File: src/app/lib/presentation/dad_jokes/favorite_dad_jokes.dart [hunk 15]
```diff
@@ -27,15 +27,17 @@ final class FavoriteDadJokesPage extends ConsumerWidget {
...
-          return ListView.builder(
-            itemCount: dadJokes.length,
-            itemBuilder: (context, index) {
-              final dadJoke = dadJokes[index];
-              return DadJokeListItem(
-                key: Key(dadJoke.id),
-                dadJoke: dadJoke,
-              );
-            },
+          return Container(
```
Comment (takla21) [inline src/app/lib/presentation/dad_jokes/favorite_dad_jokes.dart:30]: Isn't there a parameter key in the ListView builder ctor? That's a shame you need to wrap it into another widget just for tests

File: src/app/lib/business/mocking/mocking_manager.dart [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+  @override
+  Future setIsMockingEnabled(bool isMockingEnabled) async {
+    await _mockingRepository.setMocking(isMockingEnabled);
+
+     this.isMockingEnabled = isMockingEnabled;
+    hasConfigurationBeenChanged = _checkIfChangesHasBeenMade();
+  }
+
+  bool _checkIfChangesHasBeenMade() =>
```
Comment (takla21) [inline src/app/lib/business/mocking/mocking_manager.dart:48]: I don't see the point of this method here, especially if it's called only once.

File: src/app/lib/business/mocking/mocking_manager.dart [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+  _MockingManager(this._mockingRepository);
+
+  @override
+  bool hasConfigurationBeenChanged = false;
+
+  @override
+  late bool isMockingEnabled;
+    
+  @override
+  Future initialize() async {
```
Comment (takla21) [inline src/app/lib/business/mocking/mocking_manager.dart:35]: Should we use `Future` or `Future<void>` here? I used the latter on code I've contributed in this project and to be honest I'm not sure which one is better. @Soap-141 @jeanplevesque

Comment (Marc-Antoine-Soucy) [reply src/app/lib/business/mocking/mocking_manager.dart:35]: I looked it up because you mentioned it, and apparently, Furure (without void) can return any time, sort of like Future<dynamic>, I guess Future<void> would be better if we know we wont return anything

**Review by Marc-Antoine-Soucy — APPROVED** [review]

File: src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart (hunk 2 above)
Comment (Lee31416) [reply src/app/lib/presentation/diagnostic/expanded_diagnostic_page.dart:32]: yes

File: doc/Testing.md (hunk 5 above)
Comment (Lee31416) [reply doc/Testing.md:84]: it does exist it's located in dad_jokes_data_mock.dart

File: doc/Testing.md (hunk 6 above)
Comment (Lee31416) [reply doc/Testing.md:131]: The name of the package is integration_test

Comment (Lee31416) [reply doc/Testing.md:131]: image

File: doc/Testing.md (hunk 8 above)
Comment (jeanplevesque) [reply doc/Testing.md:171]: Also, for big paragraphs, I suggest you do 1 sentence per line. It doesn't affect the rendered markdown, but helps reviewing changes.

File: src/app/lib/business/mocking/mocking_manager.dart (hunk 17 above)
Comment (jeanplevesque) [reply src/app/lib/business/mocking/mocking_manager.dart:35]: > Future<void>: It's good practice to specify the return type even if it's void for better code clarity.

From ChatGPT

File: doc/Testing.md (hunk 6 above)
Comment (jeanplevesque) [reply doc/Testing.md:131]: Oh ok, I got confused because of the url.

File: src/app/lib/business/mocking/mocking_manager.dart (hunk 17 above)
Comment (Soap-141) [reply src/app/lib/business/mocking/mocking_manager.dart:35]: Refactor the whole application then.

Comment (Soap-141) [reply src/app/lib/business/mocking/mocking_manager.dart:35]: I would do it here on a separate commit, but I agree with both solutions.

File: src/app/lib/presentation/dad_jokes/favorite_dad_jokes.dart (hunk 15 above)
Comment (Lee31416) [reply src/app/lib/presentation/dad_jokes/favorite_dad_jokes.dart:30]: there is a key parameter but for some unknow reason it doesn't enable us to find it's descendants with the finder

File: src/app/test/business/dad_jokes_service_test.dart [hunk 18]
```diff
@@ -71,7 +74,7 @@ void main() {
   group("favorites", () {
     test('Get all favorite jokes', () async {
       // Arrange
-      var mockedJokesList = getFavoriteDadJokesList()
+      var mockedJokesList = getMockedFavoriteDadJokesList()
```
[the remark this answers is not in the document]
Comment (Lee31416) [reply src/app/test/business/dad_jokes_service_test.dart]: yes

File: src/app/integration_test/dad_jokes_page_test.dart [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+import 'package:app/presentation/dad_jokes/dad_joke_list_item.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:integration_test/integration_test.dart';
+
+Future<void> main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  await initializeComponents(overrideIsMocked: true);
```
[the remark this answers is not in the document]
Comment (Lee31416) [reply src/app/integration_test/dad_jokes_page_test.dart]: could be a suggestion cuz this pr is pretty heavy in changes already imo

[the remark this answers is not in the document]
Comment (jeanplevesque) [reply src/app/integration_test/dad_jokes_page_test.dart]: Yeah let's evaluate this later.

File: src/app/integration_test/dad_jokes_page_test.dart [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+      await tester.pumpAndSettle();
+
+      var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+      var dadJokes = find.descendant(
+        of: dadjokesContainer,
+        matching: find.byType(DadJokeListItem),
+      );
+
+      await tester.tap(dadJokes.first);
```
Comment (jeanplevesque) [inline src/app/integration_test/dad_jokes_page_test.dart:53]: I think this test would be better if the favorites were already present (so that you don't add to add to favorites first).

File: src/app/lib/main.dart [hunk 21]
```diff
@@ -112,19 +120,43 @@ void _registerHttpClient() {
 }
 
 /// Registers the repositories.
-void _registerRepositories() {
+void _registerRepositories({bool? overrideIsMocked}) {
```
Comment (jeanplevesque) [inline src/app/lib/main.dart:123]: You should also change overrideIsMocked to isMocked here too.

File: src/app/lib/main.dart [hunk 22]
```diff
@@ -153,8 +185,8 @@ void _registerRepositories() {
 void _registerServices() {
   GetIt.I.registerSingleton(
     DadJokesService(
-      GetIt.I.get<DadJokesRepository>(),
-      GetIt.I.get<FavoriteDadJokesRepository>(),
+      GetIt.I.get<DadJokesRepositoryDecorator>(),
```
Comment (jeanplevesque) [inline src/app/lib/main.dart:188]: Here I think you should resolve the interfaces and not the concrete implementations.

File: src/app/lib/access/mocking/mocking_repository.dart [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+
+/// Defines the repository for mocking data throughout the app.
+abstract interface class MockingRepository {
+  factory MockingRepository() = _MockingRepository;
+
+  /// Gets whether the mocking is enabled.
+  Future<bool> isMockingEnabled();
+
+  /// Sets the mocking state.
+  Future setMocking(bool enabled);
```
Comment (Soap-141) [inline src/app/lib/access/mocking/mocking_repository.dart:11]: Future<void>

File: src/app/lib/business/environment/environment.dart [hunk 24]
```diff
@@ -2,12 +2,14 @@
...
   development,
   staging,
+  generatedStaging,
   production;
 
   @override
   String toString() => switch (this) {
         development => 'Development',
         staging => 'Staging',
+        generatedStaging => 'Generated_Staging',
```
Comment (Soap-141) [inline src/app/lib/business/environment/environment.dart:12]: Wait why is this required?

File: src/app/lib/business/mocking/mocking_manager.dart [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+  factory MockingManager(MockingRepository mockingRepository) = _MockingManager;
+
+  /// Gets whether file logging is enabled.
+  bool get isMockingEnabled;
+
+  /// Gets whether mocking configuration been changed via either [setIsConsoleLoggingEnabled].
+  bool get hasConfigurationBeenChanged;
+
+  /// Sets whether mocking should be enabled on next app launch.
+  Future setIsMockingEnabled(bool isMockingEnabled);
```
Comment (Soap-141) [inline src/app/lib/business/mocking/mocking_manager.dart:14]: Future<void> in this file

File: src/app/lib/business/environment/environment.dart (hunk 24 above)
Comment (takla21) [reply src/app/lib/business/environment/environment.dart:12]: It's due to this https://github.com/nventive/FlutterApplicationTemplate/blob/da4fd805b84230907291fca2335523c1d662b967/build/azure-pipelines.yml#L41

File: src/app/.env.generated_staging [hunk 26]
```diff
@@ -0,0 +1,8 @@
+IS_CONSOLE_LOGGING_ENABLED=false
```
Comment (takla21) [inline src/app/.env.generated_staging:1]: I would avoid create this file and find a way that generated_staging uses the staging file instead

File: src/app/lib/business/environment/environment.dart (hunk 24 above)
Comment (takla21) [inline src/app/lib/business/environment/environment.dart:12]: We need to make sure that this isn't part of the generated template code. It can be acheive by using `#if false`

File: src/app/.env.generated_staging (hunk 26 above)
Comment (Soap-141) [reply src/app/.env.generated_staging:1]: The env we pass should still be Staging, we prob just need to change the value in azure-pipeline.yml

File: src/app/lib/business/environment/environment_manager.dart [hunk 27]
```diff
@@ -30,6 +30,7 @@ final class _EnvironmentManager implements EnvironmentManager {
   final Map<Environment, String> _environmentFileNames = {
     Environment.development: '.env.dev',
     Environment.staging: '.env.staging',
+    Environment.generatedStaging: '.env.generated_staging',
```
Comment (takla21) [inline src/app/lib/business/environment/environment_manager.dart:33]: it should use `.env.staging`

File: src/app/lib/business/environment/environment.dart (hunk 24 above)
Comment (Soap-141) [reply src/app/lib/business/environment/environment.dart:12]: Not implemetend, only in build folder for now, see `RemoveCommentedOutBlocks` in powershell, will refactor it in the CLI tool.

File: src/app/.env.generated_staging (hunk 26 above)
Comment (takla21) [reply src/app/.env.generated_staging:1]: if you change the value in the azure-pipeline, it will cause conflicts when creating the artifacts

Comment (takla21) [reply src/app/.env.generated_staging:1]: How I see it when I helped @Lee31416, it's either we do:
- tweak the pipeline logic to set the environment & modify the pipeline logic to have a different artifact name
- modify the code to accept `generated_staging`

Comment (Soap-141) [reply src/app/.env.generated_staging:1]: Yeah, right, not a good thing

Comment (Soap-141) [reply src/app/.env.generated_staging:1]: Artifact name could be a parameter

Comment (Soap-141) [reply src/app/.env.generated_staging:1]: image

Comment (jeanplevesque) [reply src/app/.env.generated_staging:1]: generated_staging c'est pas un environment. Ça doit être staging. Le ArtifactName peut être différent de l'environment et pourrait donc être generated_staging

File: src/app/lib/main.dart (hunk 22 above)
Comment (Lee31416) [reply src/app/lib/main.dart:188]: Doing this makes it not work anymore, plus on .net we register the decorator 
image

File: src/app/integration_test/dad_jokes_page_test.dart (hunk 20 above)
Comment (Lee31416) [reply src/app/integration_test/dad_jokes_page_test.dart]: I changed it so there's already 1 mocked favorites and it just removes it for the test

File: src/app/lib/business/environment/environment.dart (hunk 24 above)
Comment (Lee31416) [reply src/app/lib/business/environment/environment.dart]: Removed the change in favor of doing the fix on the pipeline instead

File: src/app/.env.generated_staging (hunk 26 above)
Comment (Lee31416) [reply src/app/.env.generated_staging]: Applied the change for the artifact name instead

File: src/app/integration_test/dad_jokes_page_test.dart [hunk 28]
```diff
@@ -0,0 +1,103 @@
...
+
+      var favJokesContainer = find.byKey(const Key('FavoriteJokesContainer'));
+      var favJokes = find.descendant(
+        of: favJokesContainer,
+        matching: find.byType(DadJokeListItem),
+      );
+
+      // Act
+      await tester.tap(favJokes.first);
+      await tester.pumpAndSettle();
```
Comment (takla21) [inline src/app/integration_test/dad_jokes_page_test.dart:57]: nitpick: to be consistent, shouldn't the following comment ` // Without this the dadjokesContainer isn't there yet.` be there?

File: src/app/lib/business/mocking/mocking_manager.dart [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+  /// Gets whether file logging is enabled.
+  bool get isMockingEnabled;
+
+  /// Gets whether mocking configuration been changed via either [setIsConsoleLoggingEnabled].
+  bool get hasConfigurationBeenChanged;
+
+  /// Sets whether mocking should be enabled on next app launch.
+  Future<void> setIsMockingEnabled(bool isMockingEnabled);
+
+  /// Create an instancee of [Logger].
```
Comment (takla21) [inline src/app/lib/business/mocking/mocking_manager.dart:16]: the comment here doesn't make sense

**Review by takla21 — APPROVED** [review]

File: src/app/integration_test/dad_jokes_page_test.dart (hunk 28 above)
Comment (Lee31416) [reply src/app/integration_test/dad_jokes_page_test.dart:57]: Not really because here it is used to trigger a frame change and where I put the comments it is used to wait for the app to be fully loaded before moving on

File: build/stage-build.yml [hunk 30]
```diff
@@ -72,13 +75,14 @@ jobs:
     pool:
       vmImage : $(windowsHostedAgentImage)
     variables:
-      ArtifactName: $(TestsArtifactName)_${{ parameters.applicationEnvironment }}
+      ArtifactName: ${{ parameters.artifactName }}
```
Comment (Soap-141) [inline build/stage-build.yml:78]: Make sure you are using it though.

**Review by Soap-141 — APPROVED** [review]

File: build/stage-build.yml (hunk 30 above)
Comment (Lee31416) [reply build/stage-build.yml:78]: Am I not ?

Comment (Soap-141) [reply build/stage-build.yml:78]: Don,t know, did not see the change. Might just be me, confirm please.

File: src/app/test/business/dad_jokes_service_test.dart [hunk 31]
```diff
@@ -1,49 +1,51 @@
...
           .map((dadJoke) => jsonEncode(dadJoke))
           .toList()
     });
 
-    provideDummy(mockedDadJokeResponse);
-    dadJokesRepository = MockDadJokesRepository();
-    when(dadJokesRepository.getDadJokes()).thenAnswer(
-      (_) async => mockedDadJokeResponse,
+    mockingRepository = MockingRepository();
+    dadJokesRepository = DadJokesRepositoryDecorator(
```
Comment (jeanplevesque) [inline src/app/test/business/dad_jokes_service_test.dart:33]: Since this is a test, I guess you don't really need the decorator (and especially the repository with a Dio instance) and should instead rely only on the mock implementation?

File: build/stage-build.yml (hunk 30 above)
Comment (Lee31416) [reply build/stage-build.yml:78]: it is

File: src/app/test/business/dad_jokes_service_test.dart (hunk 31 above)
Comment (Lee31416) [reply src/app/test/business/dad_jokes_service_test.dart]: I changed it back to mockito instead of the decorator for simplicity

File: src/app/lib/access/dad_jokes/mocks/dad_jokes_list_mock.dart [hunk 32]
```diff
@@ -0,0 +1,20 @@
+import 'package:app/access/dad_jokes/data/dad_joke_content_data.dart';
+import 'package:app/access/dad_jokes/mocks/dad_jokes_data_mock.dart';
+
+List<DadJokeContentData> getDadJokesList() {
```
Comment (jeanplevesque) [inline src/app/lib/access/dad_jokes/mocks/dad_jokes_list_mock.dart:4]: Shouldn't this be `getMockedDadJokesList` to be consistent with the other method?

File: src/app/lib/business/mocking/mocking_manager.dart [hunk 33]
```diff
@@ -0,0 +1,48 @@
+import 'package:app/access/mocking/mocking_repository.dart';
+
+/// Service that manage everything related to mocking.
```
Comment (jeanplevesque) [inline src/app/lib/business/mocking/mocking_manager.dart:3]: manages

**Review by jeanplevesque — APPROVED** [review]
